pkg/controllers/scheduling/scheduling_controller.go

```diff
@@ -0,0 +1,234 @@
+) factory.Controller {
+	// build controller
+	c := schedulingController{
+		clusterClient:           clusterClient,
+		clusterLister:           clusterLister,
+		clusterSetLister:        clusterSetLister,
+		clusterSetBindingLister: clusterSetBindingLister,
+		placementLister:         placementInformer.Lister(),
+		placementDecisionLister: placementDecisionInformer.Lister(),
+		scheduleFunc:            schedule,
+		rateLimiter: workqueue.NewMaxOfRateLimiter(
+			workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 120*time.Second),
+			&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+		),
+	}
+
+	return factory.New().
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			key, _ := cache.MetaNamespaceKeyFunc(obj)
+			return key
+		}, placementInformer.Informer()).
+		WithFilteredEventsInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			labels := accessor.GetLabels()
+			placementName := labels[placementLabel]
+			return fmt.Sprintf("%s/%s", accessor.GetNamespace(), placementName)
+		}, func(obj interface{}) bool {
+			accessor, err := meta.Accessor(obj)
+			if err != nil {
+				return false
+			}
+			labels := accessor.GetLabels()
+			if _, ok := labels[placementLabel]; ok {
+				return true
+			}
+			return false
+		}, placementDecisionInformer.Informer()).
+		// TODO: monitor more resources, like clusters, clustersets and clustersetbindings
+		WithSync(c.sync).
+		ResyncEvery(ResyncInterval).
+		ToController("SchedulingController", recorder)
+}
+
+func (c *schedulingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	queueKey := syncCtx.QueueKey()
+	namespace, name, err := cache.SplitMetaNamespaceKey(queueKey)
+	if err != nil {
+		// ignore placement whose key is not in format: namespace/name
+		utilruntime.HandleError(err)
+		return nil
+	}
+
+	klog.V(4).Infof("Reconciling placement %q", queueKey)
+	placement, err := c.placementLister.Placements(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		// no work if placement is deleted
+		c.rateLimiter.Forget(queueKey)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// no work if placement is deleting
+	if !placement.DeletionTimestamp.IsZero() {
+		c.rateLimiter.Forget(queueKey)
+		return nil
+	}
+
+	// get available clusters for this placement
+	clusters, err := c.getAvailableClusters(placement)
+	if err != nil {
+		return err
+	}
+
+	// schedule placement with scheduler
+	scheduleResult, err := c.scheduleFunc(ctx, placement, clusters, c.clusterClient, c.placementDecisionLister)
+	if err != nil {
+		return err
+	}
+
+	// update placement status if necessary
+	err = c.updateStatus(ctx, placement, scheduleResult.scheduled, scheduleResult.unscheduled)
+	if err != nil {
+		return err
+	}
+
+	// requeue the placement if there is still unscheduled cluster decision
+	if scheduleResult.unscheduled > 0 {
+		syncCtx.Queue().AddAfter(queueKey, c.rateLimiter.When(queueKey))
```

Comment:
you may want to consider a fixed unit of time here.  Maybe 10 minutes.  If known inputs change, this will re-evaluate earlier, so this requeue is merely a backstop.

Reply:
Since the controller will resync in two minutes, it becomes unnecessary to requeue a placement with unscheduled decision. I removed the rate limiter from the controller.